src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -76,16 +74,32 @@ void AddRedirections(Dictionary<string, SourceInfo<string>> items, bool checkRed
                     if (type != ContentType.Page)
                     {
                         errors.Add(Errors.RedirectionInvalid(redirectUrl, path));
+                        continue;
                     }
-                    else if (checkRedirectUrl && !redirectUrl.Value.StartsWith('/'))
-                    {
-                        errors.Add(Errors.RedirectionUrlInvalid(redirectUrl));
-                    }
-                    else if (checkRedirectUrl && !redirectUrls.Add(redirectUrl))
+
+                    Document redirect = Document.Create(docset, pathToDocset, redirectUrl);
+                    if (redirectDocumentId)
                     {
-                        errors.Add(Errors.RedirectionUrlConflict(redirectUrl));
+                        var linkType = UrlUtility.GetLinkType(redirectUrl);
+                        string mutableRedirectUrl = redirectUrl;
+                        if (linkType == LinkType.RelativePath)
+                        {
+                            mutableRedirectUrl = PathUtility.NormalizeFile(Path.Combine(Path.GetDirectoryName(redirect.SiteUrl), redirectUrl));
```

Comment:
Dependency.Resolve

Reply:
The `DependencyResolver` is initialized inside `context`, after docset is created.